python/lsst/ts/fiberspectrograph/csc.py

```diff
@@ -260,7 +262,10 @@ async def save_data(self, spec_data):
         image_sequence_array, data = await self.image_service_client.get_next_obs_id(
             num_images=1
         )
-        hdulist[0].header["OBSID"] = image_sequence_array[0]
+        hdulist[0].header["OBSID"] = data[0]
+        hdulist[0].header["TELCODE"] = self.config.location
+        hdulist[0].header["SEQNUM"] = data[0].split("_")[-1]
```

Comment:
SEQNUM is meant to be an integer so this is going to have to have an `int()` around it.

Reply:
ok